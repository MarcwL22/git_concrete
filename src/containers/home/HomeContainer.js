import React, { Component } from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
// Actions
import { homeOperations } from './duck';
// Components
import HomeLogo from './components/HomeLogo';
import HomeSearchForm from './components/HomeSearchForm';

const HomeSection = styled.section`
  min-height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
`;

export class HomeContainer extends Component {
  state = { userInput: '' };

  onChangeText = e => {
    const { value } = e.target;
    this.setState({ userInput: value });
    clearTimeout(this.timer);
    this.timer = setTimeout(() => {
      this.props.fetchUser(value);
    }, 200);
  };

  searchFormSubmit = e => {
    e.preventDefault();
    this.props.history.push(`/${this.state.userInput}`);
  };

  render() {
    const { userInput } = this.state;
    const { loading, error, user } = this.props;
    console.log('this.props: ', this.props);
    return (
      <HomeSection>
        <HomeLogo />
        <HomeSearchForm
          error={error}
          loading={loading}
          onChangeText={this.onChangeText}
          result={user}
          value={userInput}
          onSubmit={this.searchFormSubmit}
        />
      </HomeSection>
    );
  }
}

const mapStateToProps = ({ homeReducer }) => {
  return { ...homeReducer };
};

export default connect(
  mapStateToProps,
  { ...homeOperations }
)(HomeContainer);
