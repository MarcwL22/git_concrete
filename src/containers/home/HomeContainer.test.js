import React from 'react';
import { shallow, mount } from 'enzyme';
import { HomeContainer } from './HomeContainer';

describe('HomeContainer', () => {
  let props = {
    loading: false,
    error: false,
    user: {}
  };
  let homeContainer = shallow(<HomeContainer {...props} />);
  it('should renders properly', () => {
    expect(homeContainer).toMatchSnapshot();
  });
});
