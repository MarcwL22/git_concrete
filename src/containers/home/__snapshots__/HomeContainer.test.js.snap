// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeContainer should renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomeContainer
    error={false}
    loading={false}
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <HomeLogo />,
        <HomeSearchForm
          error={false}
          loading={false}
          onChangeText={[Function]}
          onSubmit={[Function]}
          result={Object {}}
          value=""
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "error": false,
          "loading": false,
          "onChangeText": [Function],
          "onSubmit": [Function],
          "result": Object {},
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <HomeLogo />,
          <HomeSearchForm
            error={false}
            loading={false}
            onChangeText={[Function]}
            onSubmit={[Function]}
            result={Object {}}
            value=""
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "error": false,
            "loading": false,
            "onChangeText": [Function],
            "onSubmit": [Function],
            "result": Object {},
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
